[Zoebelle Pang] iP

zoebelle-pang opened the pull request:
# Bentley
The background color is `#ffffff` for light mode and `#000000` for dark mode.
`#0969DA`

"Your mind is for having ideas, not holding them." – David Allen (source)

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

*All you need to do is,*

1. [Download it from here.](https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/project.html)
2. Double-click it.
3. Add your tasks.
4. Let it manage your tasks for you 😉

[README guidelines for this project](ip/docs/README.md)

And it is FREE!

**Features:**

- [ ] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

![showcase](https://zoebelle-pang.github.io/ip/Ui.png)

Some input commands are:
```
todo homework
```

> [!NOTE]
> Use clear and detailed task outlines to improve how you handle your tasks. It's easier to find jobs and keep things organised in Bentley when details are clear.

Click [here](https://github.com) to visit GitHub and see some `inline code`.

```ruby
require 'redcarpet'
markdown = Redcarpet.new("Hello World!")
puts markdown.to_html
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

zoebelle-pang committed:
original

zoebelle-pang committed:
Merge branch 'nus-cs2103-AY2324S2:master' into master

zoebelle-pang committed:
Level-0

zoebelle-pang committed:
Level-1

zoebelle-pang committed:
Level-2

zoebelle-pang committed:
Level-3

zoebelle-pang committed:
Level-4

zoebelle-pang committed:
Level-5

zoebelle-pang committed:
Revert "Level-5"

This reverts commit ce111a494feab027a2f4198856b5f82e0e0c1ef6.

zoebelle-pang committed:
A-TextUiTesting

zoebelle-pang committed:
Level-5

zoebelle-pang committed:
Level-6

tsulim commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,172 @@
+    }
+
+    public void markAsDone() {
+        this.isDone = true;
```
Should the `this` keyword be used here? Perhaps you could refer to the [Java Coding Standard](https://se-education.org/guides/conventions/java/index.html) under the variables section?

tsulim commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,172 @@
+    }
+
+    public void markAsUndone() {
+        this.isDone = false;
```
Similar to the previous comment, should the `this` keyword be used here?

tsulim commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,172 @@
+    }
+}
+
 public class Duke {
```
Perhaps you could change the class name to your bot's name?

tsulim commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,172 @@
+
+    public Task(String taskDescription) {
+        this.taskDescription = taskDescription;
+        this.isDone = false;
```
I noticed the same issue in other places too.

tsulim commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,172 @@
+            try {
+                String userInput = scanner.nextLine();
+
+                if (userInput.equals("Bye")) {
+                    System.out.println("Bye. Hope to see you again soon!");
+                    break;
+                } else if (userInput.equals("List")) {
+
+                    for (int i = 0; i < tasks.size(); i++) {
+                        System.out.println((i + 1) + ". " + tasks.get(i));
+                    }
+
+                } else if (userInput.startsWith("todo")) {
```
I have only highlighted a small portion but good job on using K&R style brackets (aka [Egyptian style](https://blog.codinghorror.com/new-programming-jargon/)) throughout your code.

tsulim commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,172 @@
+                    int taskNumber = Integer.parseInt(userInput.split(" ")[1]);
+                    tasks.get(taskNumber - 1).markAsDone();
+
+                    for (int i = 0; i < tasks.size(); i++) {
+                        System.out.println((i + 1) + ". " + tasks.get(i));
```
I liked how you have added white spaces within the statements according to the [Java Coding Standard](https://se-education.org/guides/conventions/java/index.html).

tsulim commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,172 @@
+    }
+
+    public String toString() {
+        return "[" + (isDone ? "X" : " ") + "] " + taskDescription;
```
Good use of whitespaces here! 👍

tsulim reviewed (commented):
Hello there!👋

Great job overall. Most codes are in line with the Java Coding Standard and are easy to read.
Just minor inconsistencies with the Java Coding Standard need to be fixed. 🛠️ 
Keep up the good work! 💯 

Also, I noticed that your Week 3's tasks are missing, please remember to push them if you have already completed them! If you have not done so, I wish you all the best! 💪✨

zoebelle-pang committed:
Level-7

zoebelle-pang committed:
Merge branch 'branch-Level-7'

SamuelZero1 commented on src/main/java/Duke.java:
perhaps it should be while the input is not equal to "bye", rather than while(true)?

zoebelle-pang committed:
Level 8

zoebelle-pang committed:
level 8.1

zoebelle-pang committed:
level 8.2

zoebelle-pang committed:
A-MoreOOP

zoebelle-pang committed:
Revert "A-MoreOOP"

This reverts commit 94df8a9a1854929788c33ea41642ccbbc8636908.

zoebelle-pang committed:
Revert "level 8.2"

This reverts commit 29f9d66cf8891f9fa0c68de058a1327b017a90f6.

zoebelle-pang committed:
remove class file

zoebelle-pang committed:
level 8.3

zoebelle-pang committed:
level 8.4

zoebelle-pang committed:
A-MoreOOP

zoebelle-pang committed:
A-Packages

zoebelle-pang committed:
Merge branch 'add-gradle-support'

zoebelle-pang committed:
A-JUnit

zoebelle-pang committed:
A-JavaDoc

zoebelle-pang committed:
A-JavaDoc all

zoebelle-pang committed:
Revert "A-JavaDoc all"

This reverts commit 9ed10ba6f75ac3d6c6890b805697d033ef48f207.

zoebelle-pang committed:
Revert "Revert "A-JavaDoc all""

This reverts commit db2313f3d156b29b71109c83ed3c1825bfe165d8.

zoebelle-pang committed:
A-CodingStandard

zoebelle-pang committed:
Level-9 Add 'find' command

zoebelle-pang committed:
inc

zoebelle-pang committed:
update package name

zoebelle-pang committed:
Add GUI

zoebelle-pang committed:
Implement GUI

zoebelle-pang committed:
fix conflict error

zoebelle-pang committed:
fix conflict error

zoebelle-pang committed:
update readme

zoebelle-pang committed:
Improve file open handling in Storage by introducing a fileIsOpen flag for status tracking

Enhance file open handling in the Storage class to provide better feedback and prevent potential issues.
Introduce a boolean flag, fileIsOpen, to track the status of the file.

Step forward, utilize assert statements in Duke class to ensure file openness before processing user input.

zoebelle-pang committed:
A-codeQuality

zoebelle-pang committed:
update

zoebelle-pang committed:
edit listTask() to make spacing consistent

zoebelle-pang committed:
Merge pull request #2 from zoebelle-pang/branch-A-Assertions

Branch a assertions

zoebelle-pang committed:
Revert "edit listTask() to make spacing consistent"

This reverts commit eef2934a8606b25abbcc93b811c894ad958476d4.

zoebelle-pang committed:
Merge pull request #3 from zoebelle-pang/branch-A-CodeQuality

Branch a code quality

zoebelle-pang committed:
BCD-Extension

C-DetectDuplicates

Detect duplicate in todo tasks

zoebelle-pang committed:
Merge branch 'master' of https://github.com/zoebelle-pang/ip

# Conflicts:
#	data/duke.txt

zoebelle-pang committed:
A-UserGuide

zoebelle-pang committed:
final changes

zoebelle-pang committed:
DukeException

zoebelle-pang committed:
minor indentation changes

zoebelle-pang committed:
Revert "minor indentation changes"

This reverts commit 0840c85b7bce17b1528460f4b653b0bceb414831.

zoebelle-pang committed:
Fix Jar file
